src/eei.h

```diff
@@ -115,6 +119,9 @@ struct EthereumInterface : ShellExternalInterface {
   evm_message const& msg;
   std::vector<uint8_t> lastReturnData;
   ExecutionResult & result;
+  const bool debug;
+  /* FIXME */
+  // static std::ostream& DebugStream;
```

Comment:
@axic :
What I am doing here to replace HERA_DEBUG (previously defined by build options) is an ostream object reference that is defined for each instance of EthereumInterface, based on the debug flag. The issue I am having is that I want to use the NullStream struct we have defined already (the one without the large amount of useless inherited public methods), but this will not work because it is not an ostream like std::cerr. What is the best way to do this?

Reply:
@chfast can you check this?

Reply:
Can you give an example what is _not_ working?

Reply:
Nothing is broken at the moment, but I am not sure what is the best way to implement the HERA_DEBUG stream such that it works with runtime debug options.
What I do at the moment is give EthereumInterface an object reference to a stream, which is either NullStream or std::cerr depending on what value of "debug" is passed to the constructor. This doesn't work because NullStream is not an ostream and so DebugStream cannot refer to it.

The two solutions I have in mind are:
1. Implement NullStream with std::ostream as the base class and inherit a bunch of stuff we don't need
2. preprocessor hell

This is not super important because I am redoing this later, retaining compile-time options, but I would like to know how to approach this type of issue.